### PR TITLE
Add warmup_type arguments in WarmupLR and WarmupDecayLR

### PR DESCRIPTION
**Pull request by skpig**
About issue #1298 @tjruwase 
I add `warmup_type` arguments in WarmupLR and WarmupDecayLR. If `log`  is chosen(Default) , the training curve will be the same as before. If `linear`, then learning rate will increase linearly during warmup.

### CODE REVIEW

**Comment by tjruwase** [comment]
Thanks. These looks good. Could you please add some unit tests for this [here](https://github.com/microsoft/DeepSpeed/blob/master/tests/unit/test_lr_schedulers.py)?

**Comment by skpig** [comment]
When I ran `pytest --forked tests/unit/test_lr_schedulers.py`, every test failed with Error reported
`AttributeError: Can't pickle local object 'distributed_test.<locals>.dist_wrap.<locals>.dist_init'`.
I'm using MacOS with no CUDA. And I already install `pytest-forked`. Could you please help me deal with it?

**Comment by tjruwase** [comment]
@skpig, not a problem for now since the CI will also run the tests. I will investigate any failures.

**Comment by skpig** [comment]
Any problems during review? If so, please feel free to at me.

File: deepspeed/runtime/lr_schedules.py [hunk 1]
```diff
@@ -727,6 +736,13 @@ def __init__(self,
         self.max_lrs = self._format_param(self.optimizer, warmup_max_lr, "max_lr")
         self.delta_lrs = [big - small for big, small in zip(self.max_lrs, self.min_lrs)]
         self.warmup_num_steps = max(2, warmup_num_steps)
+        # Currently only support linear and log function
+        if warmup_type not in {'log', 'linear'}:
```
Comment (tjruwase) [inline deepspeed/runtime/lr_schedules.py:740]: Can we avoid hardcoded constants 'log' and 'linear' with global vars, such as WARMUP_LOG_RATE and WARMUP_LINEAR_RATE? Or any names that seem reasonable? Thanks.

Comment (skpig) [reply deepspeed/runtime/lr_schedules.py]: @tjruwase Thanks for your suggestion. I've just replaced them with global variables

**Review by tjruwase — APPROVED** [review]

**Pull request merged**